src/AbstractQuery.php

```diff
@@ -905,7 +905,7 @@ private function executeIgnoreQueryCache(
             $this->setHydrationMode($hydrationMode);
         }
 
-        if (! empty($parameters)) {
+        if ($parameters !== null && count($parameters) !== 0) {
```

Comment:
Previously, we've set even an empty `ArrayCollection`. This looks like a BC break to me.

```suggestion
        if ($parameters !== null && $parameters !== []) {
```